app/scripts/stores/dashboardStore.js

```diff
@@ -0,0 +1,290 @@
+        ancestors.forEach((ancestor) => {
+            let {id, kind} = ancestor
+            if(!this.downloadedItems.has(id) && kind !== 'dds-project') {
+                this.getItem(id, this.pathFinder(kind))
+            }
+        })
+        this.ancestorStatus.set('download', false)
+    }
+    
+    @action getAncestorsChildren(ancestors) {
+        ancestors.forEach((a) => {
+            let {id, kind} = a
+            let ancestor = this.downloadedItems.get(id)
+            if(ancestor && !ancestor.childrenIds) {
+                this.getTreeListChildren(ancestor)
+            }
+        })
+        this.ancestorStatus.set('downloadComplete', true)
+        this.ancestorStatus.set('downloadChildren', false)
+    }
+
+    @action getTreeListChildren(parent) {
+        mainStore.loading = true;
+        let parentId = parent.id
+        let path = this.pathFinder(parent.kind)
+        let page = 1
+        this.transportLayer.getChildren(parentId, path, page)
+            .then(this.checkResponse)
+            .then((response) => {
+                const results = response.json();
+                const headers = response.headers;
+                return Promise.all([results, headers]);
+            })
+            .then((json) => {
+                let results = json[0].results;
+                let headers = json[1].map;
+                let childrenIds = []
+                let folderIds = []
+                let parentsChildren = results.map((child) => {
+                    childrenIds.push(child.id)
+                    if (child.kind === 'dds-folder') folderIds.push(child.id)
+                    child.parentId = parentId
+                    if (!this.downloadedItems.has(child.id)) this.downloadedItems.set(child.id, child)
+                    return ( child );
+                });
+                let downloadedParent = this.downloadedItems.get(parentId)
+                if (downloadedParent) {
+                    downloadedParent.open = true
+                    downloadedParent.childrenDownloaded = true
+                    downloadedParent.childrenIds = childrenIds
+                    downloadedParent.folderIds = folderIds
+                    this.downloadedItems.delete(parentId)
+                    this.downloadedItems.set(parentId, downloadedParent)
+                }
+                mainStore.listItems = parentsChildren
+                this.responseHeaders = headers;
+                mainStore.loading = false;
+            }).catch(ex =>this.handleErrors(ex))
+    }
+
+    @action toggleDrawer() {
+        let drawerPosition = !this.drawer.get('open')
+        let contentStyle = { transition: 'margin-left 450ms cubic-bezier(0.23, 1, 0.32, 1)' };
+        drawerPosition ? contentStyle.marginLeft = this.drawer.get('width') : null
+        this.drawer.set('open', drawerPosition)
+        this.drawer.set('contentStyle', contentStyle)
+    }
+    
+    @action setDrawer() {
+        let width = 350;
```

Comment:
move styles to component

Reply:
I will work on this on Monday

Reply:
I approved this PR but just realized that these changes still need to be made.